als-common/shared/src/main/scala/org/mulesoft/als/common/AlsAmfElement.scala

```diff
@@ -17,5 +18,34 @@ object AlsAmfElement {
     def toObject: Option[AmfObject] = obj
 
     def toArray: Option[AmfArray] = array
+
+    def containsYPart(yPartBranch: YPartBranch): Boolean =
+      amfElement match {
+        case AmfArray(values, annotations) =>
+          annotations
+            .containsYPart(yPartBranch)
+            .getOrElse( // look inside if some value is part of the branch
+              values.exists(_.containsYPart(yPartBranch)))
+        // todo: this should work to cut early, but there are cases in which a son is not contained in
+        //  the parents lexical. This should be fixed in AMF (set as Virtual)
+        //              (annotations.isVirtual || annotations.isSynthesized) && values.exists(_.containsYPart(yPartBranch)))
+
+        case amfObject: AmfObject =>
+          (amfObject.annotations.containsYPart(yPartBranch).getOrElse(false) ||
+            amfObject.annotations.containsJsonSchemaPosition(yPartBranch).getOrElse(false)) ||
+            // look inside if some value is part of the branch
+            amfObject.fields.fields().exists { fe =>
```

Comment:
The problem with the contention mechanism is that is hiding the real problem with the AMF annotation. In fact, if this is happening, there is also a missing coverage at the test suit that checks presence of annotation at all nodes of a spec.

Reply:
in case of AmfArray there does not seem to be a test failing without this look-ahead 
in case of AmfObject we have some tests that are failing, I think it should be a specific task to look into this and check if further AMF annotation changes are required

Reply:
Ok then, could be remove the look ahead for AmfArray until we notice some problem and the tackle it ?
In order to release the merge of this branch (and stop the torture of the rebase) I think we could raise the issue with the cases that you already detected for AmfObject without lexical or virtual annotation and left a TODO for remove the look ahead when is done. (I know that a TODO does not guarantee that the code will be removed after the fix, but maybe a comment at the issue could be a good reminder)